roles/openshift_storage_ceph/tasks/ceph_deploy.yml

```diff
@@ -0,0 +1,35 @@
+---
+- name: create minimal ceph cluster
+  oc_obj:
+    namespace: ""
+    kind: pod
+    name: rook-ceph
+    state: present
+    files:
```

Comment:
```suggestion
    content:
      path: "{{ tmpdir }}/cluster.yml"
```

Reply:
Hum, can you please elaborate why? Things seem to work fine with `files`. Thanks.
Btw I'm trying to reproduce the CI error on my env.

Reply:
This is what we use in ASB. I don't think it works now:
```
TASK [openshift_storage_ceph : include ceph_deploy.yml] ************************
task path: /usr/share/ansible/openshift-ansible/roles/openshift_storage_ceph/tasks/main.yml:59
Monday 07 January 2019  14:28:02 +0000 (0:00:00.499)       0:22:26.295 ******** 
included: /usr/share/ansible/openshift-ansible/roles/openshift_storage_ceph/tasks/ceph_deploy.yml for ci-op-txk6cwc0-a2bb0-ig-m-1kqn

TASK [openshift_storage_ceph : create minimal ceph cluster] ********************
task path: /usr/share/ansible/openshift-ansible/roles/openshift_storage_ceph/tasks/ceph_deploy.yml:2
Monday 07 January 2019  14:28:02 +0000 (0:00:00.116)       0:22:26.412 ******** 
changed: [ci-op-txk6cwc0-a2bb0-ig-m-1kqn] => {"changed": true, "results": {"cmd": "/bin/oc get pod rook-ceph -o json -n ", "results": [{}], "returncode": 0, "stderr": "Error from server (NotFound): pods \"rook-ceph\" not found\n", "stdout": ""}, "state": "present"}

TASK [openshift_storage_ceph : wait for the initial ceph-mon to be ready] ******
task path: /usr/share/ansible/openshift-ansible/roles/openshift_storage_ceph/tasks/ceph_deploy.yml:11
```

It seems to check that `rook-ceph` pod exists, instead of applying files for some reason

Reply:
I have the exact same log in my working environment but in the end things are working fine, see:

```
TASK [openshift_storage_ceph : create minimal ceph cluster] ********************                                                                                                                                                                                    [2990/14458]
task path: /home/leseb/leseb/openshift-ansible/roles/openshift_storage_ceph/tasks/ceph_deploy.yml:2
Monday 07 January 2019  10:20:57 -0500 (0:00:00.180)       0:15:36.047 ********
changed: [k8s-master.example.com] => {"changed": true, "results": {"cmd": "/bin/oc get pod rook-ceph -o json -n ", "results": [{}], "returncode": 0, "stderr": "Error from server (NotFound): pods \"rook-ceph\" not found\n", "stdout": ""}, "state": "present"}

TASK [openshift_storage_ceph : wait for the initial ceph-mon to be ready] ******
task path: /home/leseb/leseb/openshift-ansible/roles/openshift_storage_ceph/tasks/ceph_deploy.yml:11
Monday 07 January 2019  10:21:00 -0500 (0:00:02.702)       0:15:38.750 ********
FAILED - RETRYING: wait for the initial ceph-mon to be ready (20 retries left).
FAILED - RETRYING: wait for the initial ceph-mon to be ready (19 retries left).
FAILED - RETRYING: wait for the initial ceph-mon to be ready (18 retries left).
FAILED - RETRYING: wait for the initial ceph-mon to be ready (17 retries left).
FAILED - RETRYING: wait for the initial ceph-mon to be ready (16 retries left).
FAILED - RETRYING: wait for the initial ceph-mon to be ready (15 retries left).
ok: [k8s-master.example.com] => {"attempts": 7, "changed": false, "results": {"cmd": "/bin/oc get deployment rook-ceph-mon-a -o json -n rook-ceph", "results": [{"apiVersion": "extensions/v1beta1", "kind": "Deployment", "metadata": {"annotations": {"deployment.kubernetes.i
o/revision": "1"}, "creationTimestamp": "2019-01-07T15:21:56Z", "generation": 1, "labels": {"app": "rook-ceph-mon", "ceph_daemon_id": "a", "mon": "a", "mon_cluster": "rook-ceph", "rook_cluster": "rook-ceph"}, "name": "rook-ceph-mon-a", "namespace": "rook-ceph", "resourceV
ersion": "2328", "selfLink": "/apis/extensions/v1beta1/namespaces/rook-ceph/deployments/rook-ceph-mon-a", "uid": "f82cfbaa-128f-11e9-a386-52540034a588"}, "spec": {"progressDeadlineSeconds": 600, "replicas": 1, "revisionHistoryLimit": 10, "selector": {"matchLabels": {"app"
: "rook-ceph-mon", "ceph_daemon_id": "a", "mon": "a", "mon_cluster": "rook-ceph", "rook_cluster": "rook-ceph"}}, "strategy": {"type": "Recreate"}, "template": {"metadata": {"creationTimestamp": null, "labels": {"app": "rook-ceph-mon", "ceph_daemon_id": "a", "mon": "a", "m
on_cluster": "rook-ceph", "rook_cluster": "rook-ceph"}, "name": "rook-ceph-mon-a", "namespace": "rook-ceph"}, "spec": {"affinity": {"nodeAffinity": {"requiredDuringSchedulingIgnoredDuringExecution": {"nodeSelectorTerms": [{"matchExpressions": [{"key": "node-role.kubernete
s.io/compute", "operator": "In", "values": ["true"]}]}]}}}, "containers": [{"args": ["--foreground", "--public-addr", "172.30.103.123:6790", "--name", "mon.a", "--mon-data", "/var/lib/rook/mon-a/data"], "command": ["ceph-mon"], "env": [{"name": "POD_NAME", "valueFrom": {"
fieldRef": {"apiVersion": "v1", "fieldPath": "metadata.name"}}}, {"name": "POD_NAMESPACE", "valueFrom": {"fieldRef": {"apiVersion": "v1", "fieldPath": "metadata.namespace"}}}, {"name": "NODE_NAME", "valueFrom": {"fieldRef": {"apiVersion": "v1", "fieldPath": "spec.nodeName
"}}}], "image": "docker.io/ceph/ceph:v13", "imagePullPolicy": "IfNotPresent", "name": "mon", "ports": [{"containerPort": 6790, "name": "client", "protocol": "TCP"}], "resources": {}, "securityContext": {"privileged": true}, "terminationMessagePath": "/dev/termination-log"
, "terminationMessagePolicy": "File", "volumeMounts": [{"mountPath": "/var/lib/rook", "name": "rook-data"}, {"mountPath": "/etc/ceph", "name": "ceph-default-config-dir"}]}], "dnsPolicy": "ClusterFirst", "initContainers": [{"args": ["ceph", "mon-init", "--config-dir=/var/l
ib/rook", "--name=a", "--port=6790", "--fsid=cf5b5c24-dee6-40aa-9603-f7c871a7e8e9"], "env": [{"name": "ROOK_PRIVATE_IP", "valueFrom": {"fieldRef": {"apiVersion": "v1", "fieldPath": "status.podIP"}}}, {"name": "ROOK_PUBLIC_IP", "value": "172.30.103.123"}, {"name": "ROOK_CL
USTER_NAME", "value": "rook-ceph"}, {"name": "ROOK_MON_ENDPOINTS", "valueFrom": {"configMapKeyRef": {"key": "data", "name": "rook-ceph-mon-endpoints"}}}, {"name": "ROOK_MON_SECRET", "valueFrom": {"secretKeyRef": {"key": "mon-secret", "name": "rook-ceph-mon"}}}, {"name": "
ROOK_ADMIN_SECRET", "valueFrom": {"secretKeyRef": {"key": "admin-secret", "name": "rook-ceph-mon"}}}, {"name": "ROOK_CEPH_CONFIG_OVERRIDE", "value": "/etc/rook/config/override.conf"}], "image": "rook/ceph:master", "imagePullPolicy": "IfNotPresent", "name": "config-init",
"resources": {}, "securityContext": {"privileged": true}, "terminationMessagePath": "/dev/termination-log", "terminationMessagePolicy": "File", "volumeMounts": [{"mountPath": "/var/lib/rook", "name": "rook-data"}, {"mountPath": "/etc/ceph", "name": "ceph-default-config-di
r"}, {"mountPath": "/etc/rook/config", "name": "rook-config-override"}]}, {"args": ["/var/lib/rook/mon-a/monmap", "--create", "--clobber", "--fsid", "cf5b5c24-dee6-40aa-9603-f7c871a7e8e9", "--add", "a", "172.30.103.123:6790"], "command": ["/usr/bin/monmaptool"], "image":
"docker.io/ceph/ceph:v13", "imagePullPolicy": "IfNotPresent", "name": "monmap-init", "resources": {}, "securityContext": {"privileged": true}, "terminationMessagePath": "/dev/termination-log", "terminationMessagePolicy": "File", "volumeMounts": [{"mountPath": "/var/lib/ro
ok", "name": "rook-data"}, {"mountPath": "/etc/ceph", "name": "ceph-default-config-dir"}]}, {"args": ["--mkfs", "--monmap", "/var/lib/rook/mon-a/monmap", "--name", "mon.a", "--mon-data", "/var/lib/rook/mon-a/data"], "command": ["ceph-mon"], "image": "docker.io/ceph/ceph:v
13", "imagePullPolicy": "IfNotPresent", "name": "mon-fs-init", "resources": {}, "securityContext": {"privileged": true}, "terminationMessagePath": "/dev/termination-log", "terminationMessagePolicy": "File", "volumeMounts": [{"mountPath": "/var/lib/rook", "name": "rook-dat
a"}, {"mountPath": "/etc/ceph", "name": "ceph-default-config-dir"}]}], "nodeSelector": {"kubernetes.io/hostname": "k8s-node0.example.com"}, "restartPolicy": "Always", "schedulerName": "default-scheduler", "securityContext": {}, "terminationGracePeriodSeconds": 30, "volume
s": [{"hostPath": {"path": "/var/lib/rook", "type": ""}, "name": "rook-data"}, {"emptyDir": {}, "name": "ceph-default-config-dir"}, {"configMap": {"defaultMode": 420, "items": [{"key": "config", "path": "override.conf"}], "name": "rook-config-override"}, "name": "rook-con
fig-override"}]}}}, "status": {"availableReplicas": 1, "conditions": [{"lastTransitionTime": "2019-01-07T15:22:03Z", "lastUpdateTime": "2019-01-07T15:22:03Z", "message": "Deployment has minimum availability.", "reason": "MinimumReplicasAvailable", "status": "True", "type"
: "Available"}, {"lastTransitionTime": "2019-01-07T15:21:56Z", "lastUpdateTime": "2019-01-07T15:22:03Z", "message": "ReplicaSet \"rook-ceph-mon-a-5b4db54f88\" has successfully progressed.", "reason": "NewReplicaSetAvailable", "status": "True", "type": "Progressing"}], "ob
servedGeneration": 1, "readyReplicas": 1, "replicas": 1, "updatedReplicas": 1}}], "returncode": 0}, "state": "list"}

TASK [openshift_storage_ceph : wait for the initial ceph-mgr to be ready] ******
task path: /home/leseb/leseb/openshift-ansible/roles/openshift_storage_ceph/tasks/ceph_deploy.yml:24
Monday 07 January 2019  10:22:07 -0500 (0:01:07.255)       0:16:46.006 ********
FAILED - RETRYING: wait for the initial ceph-mgr to be ready (20 retries left).
ok: [k8s-master.example.com] => {"attempts": 2, "changed": false, "results": {"cmd": "/bin/oc get deployment rook-ceph-mgr-a -o json -n rook-ceph", "results": [{"apiVersion": "extensions/v1beta1", "kind": "Deployment", "metadata": {"annotations": {"deployment.kubernetes.i
o/revision": "1"}, "creationTimestamp": "2019-01-07T15:22:09Z", "generation": 1, "labels": {"app": "rook-ceph-mgr", "ceph_daemon_id": "a", "instance": "a", "mgr": "a", "rook_cluster": "rook-ceph"}, "name": "rook-ceph-mgr-a", "namespace": "rook-ceph", "resourceVersion": "2
380", "selfLink": "/apis/extensions/v1beta1/namespaces/rook-ceph/deployments/rook-ceph-mgr-a", "uid": "00302e7a-1290-11e9-a386-52540034a588"}, "spec": {"progressDeadlineSeconds": 600, "replicas": 1, "revisionHistoryLimit": 10, "selector": {"matchLabels": {"app": "rook-cep
h-mgr", "ceph_daemon_id": "a", "instance": "a", "mgr": "a", "rook_cluster": "rook-ceph"}}, "strategy": {"type": "Recreate"}, "template": {"metadata": {"annotations": {"prometheus.io/port": "9283", "prometheus.io/scrape": "true"}, "creationTimestamp": null, "labels": {"app
": "rook-ceph-mgr", "ceph_daemon_id": "a", "instance": "a", "mgr": "a", "rook_cluster": "rook-ceph"}, "name": "rook-ceph-mgr-a"}, "spec": {"affinity": {"nodeAffinity": {"requiredDuringSchedulingIgnoredDuringExecution": {"nodeSelectorTerms": [{"matchExpressions": [{"key":
"node-role.kubernetes.io/compute", "operator": "In", "values": ["true"]}]}]}}}, "containers": [{"args": ["--foreground", "--id", "a"], "command": ["ceph-mgr"], "env": [{"name": "POD_NAME", "valueFrom": {"fieldRef": {"apiVersion": "v1", "fieldPath": "metadata.name"}}}, {"n
ame": "POD_NAMESPACE", "valueFrom": {"fieldRef": {"apiVersion": "v1", "fieldPath": "metadata.namespace"}}}, {"name": "NODE_NAME", "valueFrom": {"fieldRef": {"apiVersion": "v1", "fieldPath": "spec.nodeName"}}}, {"name": "ROOK_CLUSTER_NAME", "value": "rook-ceph"}], "image":
 "docker.io/ceph/ceph:v13", "imagePullPolicy": "IfNotPresent", "name": "mgr", "ports": [{"containerPort": 6800, "name": "mgr", "protocol": "TCP"}, {"containerPort": 9283, "name": "http-metrics", "protocol": "TCP"}, {"containerPort": 8443, "name": "dashboard", "protocol":
"TCP"}], "resources": {}, "terminationMessagePath": "/dev/termination-log", "terminationMessagePolicy": "File", "volumeMounts": [{"mountPath": "/var/lib/rook", "name": "rook-data"}, {"mountPath": "/etc/ceph", "name": "ceph-default-config-dir"}]}], "dnsPolicy": "ClusterFir
st", "initContainers": [{"args": ["ceph", "mgr-init", "--config-dir=/var/lib/rook", "--mgr-name=a"], "env": [{"name": "ROOK_MGR_KEYRING", "valueFrom": {"secretKeyRef": {"key": "keyring", "name": "rook-ceph-mgr-a"}}}, {"name": "ROOK_PRIVATE_IP", "valueFrom": {"fieldRef": {
"apiVersion": "v1", "fieldPath": "status.podIP"}}}, {"name": "ROOK_PUBLIC_IP", "valueFrom": {"fieldRef": {"apiVersion": "v1", "fieldPath": "status.podIP"}}}, {"name": "ROOK_MGR_MODULE_SERVER_ADDR", "valueFrom": {"fieldRef": {"apiVersion": "v1", "fieldPath": "status.podIP"
}}}, {"name": "ROOK_CEPH_VERSION_NAME", "value": "mimic"}, {"name": "ROOK_MON_ENDPOINTS", "valueFrom": {"configMapKeyRef": {"key": "data", "name": "rook-ceph-mon-endpoints"}}}, {"name": "ROOK_MON_SECRET", "valueFrom": {"secretKeyRef": {"key": "mon-secret", "name": "rook-c
eph-mon"}}}, {"name": "ROOK_ADMIN_SECRET", "valueFrom": {"secretKeyRef": {"key": "admin-secret", "name": "rook-ceph-mon"}}}, {"name": "ROOK_CEPH_CONFIG_OVERRIDE", "value": "/etc/rook/config/override.conf"}], "image": "rook/ceph:master", "imagePullPolicy": "IfNotPresent",
"name": "config-init", "resources": {}, "terminationMessagePath": "/dev/termination-log", "terminationMessagePolicy": "File", "volumeMounts": [{"mountPath": "/var/lib/rook", "name": "rook-data"}, {"mountPath": "/etc/ceph", "name": "ceph-default-config-dir"}, {"mountPath":
 "/etc/rook/config", "name": "rook-config-override"}]}], "restartPolicy": "Always", "schedulerName": "default-scheduler", "securityContext": {}, "serviceAccount": "rook-ceph-mgr", "serviceAccountName": "rook-ceph-mgr", "terminationGracePeriodSeconds": 30, "volumes": [{"em
ptyDir": {}, "name": "rook-data"}, {"emptyDir": {}, "name": "ceph-default-config-dir"}, {"configMap": {"defaultMode": 420, "items": [{"key": "config", "path": "override.conf"}], "name": "rook-config-override"}, "name": "rook-config-override"}]}}}, "status": {"availableRep
licas": 1, "conditions": [{"lastTransitionTime": "2019-01-07T15:22:18Z", "lastUpdateTime": "2019-01-07T15:22:18Z", "message": "Deployment has minimum availability.", "reason": "MinimumReplicasAvailable", "status": "True", "type": "Available"}, {"lastTransitionTime": "2019
-01-07T15:22:09Z", "lastUpdateTime": "2019-01-07T15:22:18Z", "message": "ReplicaSet \"rook-ceph-mgr-a-5b7f958f76\" has successfully progressed.", "reason": "NewReplicaSetAvailable", "status": "True", "type": "Progressing"}], "observedGeneration": 1, "readyReplicas": 1, "r
eplicas": 1, "updatedReplicas": 1}}], "returncode": 0}, "state": "list"}
```